File: Ruby_api_samples/update_group.rb
#create group

require "rubygems"
require "rest_client"
require "json"


# Need to specify group_id in url
# eg:
# site = RestClient::Resource.new("http://domain.freshdesk.com/groups/2.json","sample@freshdesk.com","test")

site = RestClient::Resource.new("http://domain.freshdesk/groups/[group_id].json","sample@freshdesk.com","test")

response = site.put({:group=>{:name=>"ne group",:description=>"new group description",:assign_time=>1200,:escalate_to=>16,:agent_list=>[16,1]}},:content_type=>"application/json")

puts "response: #{response.code} \n #{response.body} \n"

File: php_samples/update_group.php
<?php

  $email="user@company.com";
  $password = "test";
  $url = 'http://domain.freshdesk.com/groups/[group_id].json';
  // Example : 'http://domain.freshdesk.com/groups/1.json'; 
  // use https if ssl enabled for the account.

  $data = array(
    "group" => array(
      "description" => "Sample Description",
      "name" => "Sample",
      "agent_list" => "1,2,4"
    )
  );

  $update_json = json_encode($data, JSON_FORCE_OBJECT | JSON_PRETTY_PRINT);
  $header[] = "Content-type: application/json";
  
  $ch = curl_init($url);
  curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
  curl_setopt($ch, CURLOPT_HTTPHEADER, $header);
  curl_setopt($ch, CURLOPT_HEADER, false);
  curl_setopt($ch, CURLOPT_USERPWD, "$email:$password");
  curl_setopt($ch, CURLOPT_POSTFIELDS, $update_json);
  curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'PUT');
  curl_setopt($ch, CURLOPT_VERBOSE, 1);
  $response = curl_exec($ch);
  curl_close($ch);

?>
